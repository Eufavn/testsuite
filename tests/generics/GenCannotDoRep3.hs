{-# LANGUAGE DeriveGeneric       #-}

module CannotDoRep3 where

import GHC.Generics

-- Only types that can derive Functor can derive Generic1
data Simple = Simple deriving Generic1
