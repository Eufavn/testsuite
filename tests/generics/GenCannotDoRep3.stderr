
GenCannotDoRep3.hs:8:31:
    Cannot derive well-kinded instance of form `Generic1 (Simple ...)'
      Class `Generic1' expects an argument of kind `* -> *'
    In the data type declaration for `Simple'
